airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -166,9 +167,8 @@ def hook(self):
     def path_to_s3_dataset(path) -> Dataset:
         from openlineage.client.run import Dataset
 
-        path = path.replace("s3://", "")
-        split_path = path.split("/")
-        return Dataset(namespace=f"s3://{split_path[0]}", name="/".join(split_path[1:]), facets={})
+        split = urlsplit(path)
+        return Dataset(namespace=f"{split.scheme}://{split.netloc}", name=split.path, facets={})
```

Comment:
Would the path contain query parameters like `?foo=bar&x`? Not familiar with this, but using `split.path` would loose those. We might need `urlunsplit(split._replace(scheme="", netloc="")).lstrip("/")` instead (getting rid of the leading slash to match the current behaviour)